src/client/datascience/interactive-ipynb/nativeEditorCommandListener.ts

```diff
@@ -97,10 +99,11 @@ export class NativeEditorCommandListener implements IDataScienceCommandListener
         }
     }
 
-    private interruptKernel() {
-        const activeEditor = this.provider.activeEditor;
-        if (activeEditor) {
-            activeEditor.interruptKernel().ignoreErrors();
+    private interruptKernel(document: NotebookDocument | undefined) {
+        const target =
+            this.provider.editors.find((editor) => editor.document === document) ?? this.provider.activeEditor;
+        if (target) {
+            target.interruptKernel().ignoreErrors();
```

Comment:
Why not just look for an editor with the same Uri.
If. this.provider.activeEditor.uri === document.uri, then we have what we need. else just iterate through the editors & look for the Uri.

This way you don't need to expose the Document in the Editor object.

Reply:
Yeah that was the other idea I had, but I saw that `document` was already public readonly in VSCNotebook and thought adding it to the interface was cleaner. If we don't want to use `INotebookEditor` I'll make your suggested change.